### PR TITLE
Remove trailing comma from nodes lists

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today when the membership of the cluster changes we log messages that describe
the change like this:

    added {{node-1}{OPdaTIGmSxaEXXOyg3o96w}{127.0.0.1}{127.0.0.1:9301}{di},}

The trailing comma suggests there is some missing string that might contain
extra information, but in fact it's an artefact of how these messages are
constructed. This commit removes the trailing comma from these lists.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java [hunk 1]
```diff
@@ -518,8 +518,14 @@ public String shortSummary() {
                     summary.append(", ");
                 }
                 summary.append("removed {");
+                boolean first = true;
                 for (DiscoveryNode node : removedNodes()) {
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java]: I guess `Strings.collectionToCommaDelimitedString()` would do the job?

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java]: Indeed it would, thanks, see d3928c6.

**Review by tlrx — APPROVED** [review]
LGTM, I left a cosmetic comment

**Comment by DaveCTurner** [comment]
I don't see your cosmetic comment @tlrx.

File: server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodesTests.java [hunk 2]
```diff
@@ -163,6 +164,36 @@ public void testMastersFirst() {
...
+        final DiscoveryNodes nodes02Local = DiscoveryNodes.builder(nodes012).localNodeId(discoveryNodes.get(1).getId()).build();
+
+        assertThat(nodes01Local.delta(nodes0).shortSummary(), equalTo(""));
+        assertThat(nodes02Local.delta(nodes0).shortSummary(), equalTo("added {" + discoveryNodes.get(2) + "}"));
+
+        assertThat(nodes0.delta(nodes01Local).shortSummary(), equalTo("removed {" + discoveryNodes.get(1) + "}"));
+        assertThat(nodes0.delta(nodes02Local).shortSummary(), oneOf(
+            "removed {" + discoveryNodes.get(1) + "," + discoveryNodes.get(2) + "}",
+            "removed {" + discoveryNodes.get(2) + "," + discoveryNodes.get(1) + "}"));
+
```
Comment (tlrx) [inline server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodesTests.java]: nit: Extra blank line

**Pull request merged**